forms-shared/src/generator/createCombinations.ts

```diff
@@ -0,0 +1,32 @@
+type InputObject = Record<string, readonly string[]>
+
+type CombinationType<T extends InputObject> = {
+  [K in keyof T]: T[K][number]
+}
+
+export function createCombinations<T extends InputObject, R>(
```

Comment:
comment on when this is used / useful, with example

Reply:
Yes, I want to replace it with some library, but need to get this to master now.